third-party/linshare/src/main/java/org/apache/james/linshare/LinshareBlobExportMechanism.java

```diff
@@ -75,8 +76,14 @@ private void uploadAndShare(MailAddress mailAddress, File tempFile, String expla
             .build());
     }
 
-    private File createTempFile(BlobId blobId, Optional<FileExtension> fileExtension) throws IOException {
+    private File createTempFile(BlobId blobId, Optional<String> fileCustomPrefix,
+                        Optional<FileExtension> fileExtension) throws IOException {
         String suffix = fileExtension.map(FileExtension::asFileSuffix).orElse("");
-        return File.createTempFile(blobId.asString() + "_", suffix);
+        String blobIdAsString = blobId.asString();
+        String fileName = fileCustomPrefix
+            .map(prefix -> prefix + blobIdAsString)
```

Comment:
A bit like the previous comment, do we need to still append a long blob id to the file name if we got the prefix defined?

Reply:
idem

Reply:
if I have two blobs but same prefix, then when uploading to linshare, one of two will be overrided, or renamed IDK, but could generate unexpected behaviors

Reply:
Are we sure about that? That if you push two files with similar names it overrides the previous one? I understand the randomness but maybe a shorter random string? I believe even if you have a readable prefix, if you put a long random string behind then it reduces the readability IMO

Reply:
The long name after the prefix is actually not really fully random, it is blobId (it takes most of the length of the name - 64 characters) + around 10 random characters. Which part would you think It can become shorter? 

Having a file with short name is better, but IDK how to make it short.

Reply:
We can generate a random string of 10 character :  `RandomStringUtils.random(10)`

Reply:
@chibenwa  any proposal ?

Reply:
random characters are automatically added at the suffix by `File.createTempFile()` I didn't intend to do it

Reply:
ok, then don't use createTempFile ?

Reply:
Then, is that what you suggested : `keep blobId in file names, and remove the random part generated by java File API`?

Reply:
createTempFile allows to create atomically a file into a directory. I guess that if we use a unique enough id, we can just create the file ourselves.
What I ask is: does we need yet another unique id for something that already has one?
The risk of using blobId is that at one point we could have two tasks writing the same file at the same time. But I think it's not the case here.
So, basically, yes, I suggest to reuse the blobId. But maybe there's arguments against that.

Reply:
Okay, no one against this